seam/plugins/org.jboss.tools.runtime.seam.detector/src/org/jboss/tools/runtime/handlers/SeamHandler.java

```diff
@@ -50,25 +51,35 @@ private static File getSeamRoot(RuntimeDefinition runtimeDefinition) {
 	}
 	
 	@Override
-	public void initializeRuntimes(List<RuntimeDefinition> runtimeDefinitions) {
-		
-		Map<String, SeamRuntime> map = new HashMap<String,SeamRuntime>();
-
-		for(RuntimeDefinition runtimeDefinition:runtimeDefinitions) {
-			if (runtimeDefinition.isEnabled()) {
-				String type = runtimeDefinition.getType();
-				if (SEAM.equals(type)) {
-					addSeam(map, runtimeDefinition,
-							runtimeDefinition.getLocation());
+	public boolean initializeRuntime(RuntimeDefinition runtimeDefinition) {
+		return initializeRuntime(runtimeDefinition, true);
+	}
+	
+	public boolean initializeRuntime(RuntimeDefinition runtimeDefinition, boolean save) {
+		if (runtimeDefinition.isEnabled()) {
+			String type = runtimeDefinition.getType();
+			if (SEAM.equals(type)) {
+				boolean ret = addSeam(runtimeDefinition,
+						runtimeDefinition.getLocation());
+				initializeRuntimes(runtimeDefinition.getIncludedRuntimeDefinitions());
```

Comment:
What if the included runtimes are not Seam ones ? This will result in no op for them? Is this wanted